python_modules/dagster/dagster/core/definitions/assets_info.py

```diff
@@ -0,0 +1,184 @@
+from typing import (
+    TYPE_CHECKING,
+    AbstractSet,
+    Callable,
+    Dict,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Set,
+    Tuple,
+)
+
+from dagster import check
+from dagster.core.definitions.events import AssetKey
+
+from .dependency import NodeHandle, NodeInputHandle, NodeOutputHandle
+from .graph_definition import GraphDefinition
+from .node_definition import NodeDefinition
+from .partition import PartitionsDefinition
+
+if TYPE_CHECKING:
+    from dagster.core.execution.context.output import OutputContext
+
+
+class AssetInfo(
```

Comment:
Could this be called `AssetOutputInfo` or something?  If I understand correctly, the non-asset_key attributes only apply to outputs.

The name `AssetInfo` feels a little too general.  E.g. I could imagine it being the type that the Python graphql client would return when queried about asset status.

Reply:
Fair point, InputDefinition does have an asset_partitions field but we don't use it for much at the moment and we'll deprecate it eventually anyway.